app/controllers/flood_risk_engine/enrollments/steps_controller.rb

```diff
@@ -30,10 +30,18 @@ def update
         end
       end
 
+      def remove_exemption
+        @enrollment = Enrollment.find(params[:id])
+        @exemption = Exemption.find(params[:exemption_id])
+        enrollment.exemptions.destroy(@exemption)
+        step_back if enrollment.exemptions.empty?
+        redirect_to step_url
+      end
+
```

Comment:
Two thoughts on alternatives to this approach: 

**1 All step-based knowledge mored to a new  CheckExemptionsForm form object**

- The 'Remove' links use a `link_to(method: :patch)` (clicking 'Remove' changes something on the server so arguably should be PATCH (or, actually, DELETE) rather than a GET). However might need to check non-js compatibility (`method: :x` on `link_to` is an unobtrusive js thing so might have to hand crank an html form around each link) 
- add a CheckExemptionsForm for this step (which would incidentally maintain the 'form object for every step' developer expectation, reducing cognitive friction. I love that phrase).
Override the form object #save.
`CheckExemptionsForm#save` would have to handle 2 situations
  1. If Remove clicked, an exemption_id is supplied - remove the exemption and redirect to same page, or redirect to add_exemptions if none left
  2. If Continue clicked, no exemption_id is passed so an assumtion made that this is the normal 'next' action - essentially nothing to do in this instance as nothing to save.

Like this, steps controller remains unchanged, and all the logic about a step is in the form object,
even if its a bit clumsy for this step. I am just concerned that otherwise we set a precedent for putting step-specific logic in the controller, offering a misleading path of least resistence for
future devs working on the code


**2 Enrollments::ExemptionsController**

Even better maybe is another alternative: to add an `Enrollments::ExcemptionsController` with a single `#delete` action,  and the Remove links hits this - which is a bit more HTTP compliant 
and RESTful than a PATCH for this case. 
However this will spread the step logic across an extra controller - not sure if that's more confusing or not